packages/rum-core/src/domain/rumEventsCollection/view/viewMetrics/trackScrollMetrics.ts

```diff
@@ -18,11 +18,28 @@ export function trackScrollMetrics(
   configuration: RumConfiguration,
   viewStart: ClocksState,
   callback: (scrollMetrics: ScrollMetrics) => void,
+  scrollHeightObservable: Observable<number>,
   getScrollValues = computeScrollValues
 ) {
   let maxDepth = 0
+
+  const subscription = scrollHeightObservable.subscribe((scrollHeight) => {
+    const { scrollDepth, scrollTop } = getScrollValues()
+
+    const now = relativeNow()
+    const maxDepthTime = elapsed(viewStart.relative, now)
+    maxDepth = scrollDepth
+    callback({
+      maxDepth,
+      maxDepthScrollHeight: scrollHeight,
+      maxDepthTime,
+      maxDepthScrollTop: scrollTop,
+    })
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏You could factorise this code with the on in the `handleScrollEvent`